lib/madwf_transfer.cu

```diff
@@ -0,0 +1,94 @@
+#include <color_spinor_field.h>
+#include <device_vector.h>
+#include <tunable_nd.h>
+#include <madwf_transfer.h>
+#include <kernels/madwf_transfer.cuh>
+#include <madwf_ml.h>
+#include <uint_to_char.h>
+
+namespace quda
+{
+  namespace madwf_ml
+  {
+
+    template <class storage_t> class transfer_5D_wrapper : public TunableKernel3D
+    {
+      ColorSpinorField &out;
+      const ColorSpinorField &in;
+      const MadwfAcc::transfer_float *wm_p;
+      bool dagger;
+      using matrix_t = typename transfer_5D_mapper<MadwfAcc::transfer_float, 4, 3, MadwfAcc::transfer_t>::type;
+
+      unsigned int sharedBytesPerThread() const { return 0; }
+
+      unsigned int sharedBytesPerBlock(const TuneParam &) const { return out.X(4) * in.X(4) * sizeof(matrix_t); }
+
+      unsigned int minThreads() const { return out.VolumeCB() / out.X(4); }
+
+    public:
+      transfer_5D_wrapper(ColorSpinorField &out, const ColorSpinorField &in, const MadwfAcc::transfer_float *wm_p,
+                          bool dagger) :
+        TunableKernel3D(out, out.X(4), out.SiteSubset()), out(out), in(in), wm_p(wm_p), dagger(dagger)
+      {
+        TunableKernel2D_base<false>::resizeStep(out.X(4)); // Ls must be contained in the block
+
+        char in_str[16];
+        char out_str[16];
+        strcat(aux, ",");
+        i32toa(in_str, in.X(4));
+        strcat(aux, in_str);
+        strcat(aux, "->");
```

Comment:
I think I'd prefer a `,` instead of an `->` here, just because that's a convention we use elsewhere, even in MG

Reply:
https://github.com/lattice/quda/pull/1061/commits/5cce447990c8ea085125244fea58154e4271dc8a